RecoLocalCalo/CaloTowersCreator/src/CaloTowersCreationAlgo.cc

```diff
@@ -953,6 +953,13 @@ void CaloTowersCreationAlgo::convert(const CaloTowerDetId& id, const MetaTower&
 	    //  towerP4 += CaloTower::PolarLorentzVector(E_had_tot*hadPf, hadPoint.eta(), hadPoint.phi(), 0); 
 	    // }
 	  }
+          if (E_had_tot==0 && E_em==0 && !theHOIsUsed && E_outer>0) { //extra-fallback for deactivated HO and zero E_em and E_had deposits to avoid nan towers
+	    auto lP4 = hadPoint.basicVector().unit();
+	    lP4[3] = 1.f;  // energy
+	    lP4 *=1e-9; // set to tiny pt to avoid nan
+	    towerP4 +=lP4;
```

Comment:
Hi Slava,

thanks for the comment! 
So, if I understand correctly you would suggest to scrap the above and put something like
CaloTower::PolarLorentzVector(**1e-9**,hadPoint.eta(), hadPoint.phi(),0)
here instead -->
    if unlikely ( (towerP4[3]==0) & (E_outer>0)  ) {
      collection.emplace_back(id, E_em, E_had, E_outer, -1, -1, CaloTower::PolarLorentzVector(0,hadPoint.eta(), hadPoint.phi(),0),  emPoint, hadPoint);
    } else {
?